private/pkg/manifest/manifest.go

```diff
@@ -139,6 +140,15 @@ func (m *Manifest) Paths() []string {
 	return paths
 }
 
+// IteratePaths invokes a function for all the paths in the manifest, passing
+// the path and its digest. The order in which the paths are iterated is not
+// guaranteed.
+func (m *Manifest) IteratePaths(f func(string, Digest)) {
```

Comment:
The paths function doesn't look too expensive - why are we introducing this variant? Is it to return the pre-calculated digest for each path?

Reply:
Is not because of how expensive it is, but because we have some code paths in which we want the digests, not only the paths, so we can drop unnecessary presence checkers. I'll change a few of the code paths in this same PR as well to make it more clear.